### PR TITLE
Add glossaries option

### PR DESCRIPTION
**Pull request by Idaapayo**
This pull request facilitates adding a glossaries option to json-autotranslate
To test this, you can copy the i18n folder into the root of the json-autotranslate. Then provide default values in the options in `index.ts` file. That is the input file, the matcher, the service and glossaries which should be a folder containing the glossary files.
The files in the glossaries folder should all be json files and the fileName should be in the format `sourceLanguage-targetLanguage.json`. The file's content should have the key in the source language and the value in the target language.
When translating, the keys in the glossaries will always be translated the same.
I am not sure how to run this without providing the default values.

### CODE REVIEW

File: package.json
Comment (leolabs) [inline package.json:11]: Afaik, the `postinstall` hook is run after running `npm i`, so adding it to the command seems redundant. What made you add this to the hook?

File: src/index.ts
Comment (leolabs) [inline src/index.ts:60]: By default, this scripts looks for locale files in the current working directory (see line 30), so the default value of this option might clash with that. It might make sense to not supply a default value here to avoid that.

File: src/services/deepl.ts
Comment (leolabs) [inline src/services/deepl.ts:75]: Let's leave the white space here for better readability.

File: src/services/deepl.ts
Comment (leolabs) [inline src/services/deepl.ts:140]: Let's leave the white-space here for better readability.

File: src/services/deepl.ts
Comment (leolabs) [inline src/services/deepl.ts:164]: Let's leave the white-space here for better readability.

File: src/services/deepl.ts
Comment (leolabs) [inline src/services/deepl.ts:186]: Let's leave the white-space here for better readability.

File: src/services/deepl.ts
Comment (leolabs) [inline src/services/deepl.ts:154]: Let's leave the white-space here for better readability.

**Review by leolabs — CHANGES_REQUESTED** [review]
Thank you for your PR, that looks like a great addition! I have a few questions and feedback before merging, see below. Could you also add the new configuration options to the README along with a few sentences on how to use them?

**Comment by steventilator** [comment]
@leolabs Sorry, I wasn't aware that my colleague already created a PR for this. I made many of these changes in order to make it work in our fork without directly creating a PR. I'll update it.

File: src/index.ts
Comment (steventilator) [reply src/index.ts:60]: Yes, we added this while developing because we couldn't figure out how to actually call the script and pass values to it. Simply providing a default, however, worked. When developing, how do I call the script? When running `npm run start` and passing the options there, they are not picked up.

Comment (steventilator) [reply src/index.ts]: Thanks, @leolabs, this worked! The code now successfully uses glossaries to translate with DeepL. Could you do another review when you get the chance? I can't request another review. Probably because it wasn't me who created the initial PR.

**Comment by steventilator** [comment]
Great, thanks! How soon is this usually published to NPM, @leolabs?

**Comment by leolabs** [comment]
Hey @steventilator, I just published a new version with this PR.

**Comment by carnivean** [comment]
Not sure whether it is related to this featute, but the following line, crashes in the createGlossaryFromFile if there is no glossary:

`json-autotranslate -i locales -m i18next -s deepl-free -c myapikey`

According to the documentation it should work or did something else change as well?

**Comment by steventilator** [comment]
I'll take a look.

**Comment by steventilator** [comment]
[This PR](https://github.com/leolabs/json-autotranslate/pull/109) should fix the bug.

**Pull request merged**
